src/components/MultiSelectCombobox/UncontrolledMultiSelectCombobox.tsx

```diff
@@ -0,0 +1,172 @@
+import noop from 'lodash.noop';
+import React, { useCallback, useEffect, useRef, useState } from 'react';
+import type { ComboboxOption } from './Combobox.types';
+import ComboboxItem from './ComboboxItem';
+import ComboboxItems from './ComboboxItems';
+import ComboboxSelection from './ComboboxSelection';
+import ComboboxSelections from './ComboboxSelections';
+import FilteredComboboxItems from './FilteredComboboxItems';
+import MultiSelectCombobox from './MultiSelectCombobox';
+
+type OptionComparisonKey = 'id' | 'value';
+
+interface UncontrolledSelectComboProps<T extends ComboboxOption> {
+  allowCreation?: boolean;
+  closeOnSelect?: boolean;
+  filterOptions?: boolean;
+  onChange?: (selections: ComboboxOption[]) => void;
+  onCreateOption?: (newOptionText: string) => ComboboxOption;
+  options: T[];
+  optionComparisonKey?: OptionComparisonKey;
+}
+
+function isOptionSelected<T extends ComboboxOption>(
+  option: T,
+  selected: T[],
+  comparisonKey: OptionComparisonKey = 'value'
+) {
+  return selected.find((selection) => selection[comparisonKey] === option[comparisonKey]);
+}
+
+function isOptionFiltered<T extends ComboboxOption>(option: T, filterText: string) {
+  return filterText === '' || option.label.toLowerCase().includes(filterText.toLowerCase());
+}
+
+const defaultProps = {
+  allowCreation: false,
+  closeOnSelect: false,
+  filterOptions: true,
+  onChange: noop,
+  optionComparisonKey: 'value' as OptionComparisonKey,
```

Comment:
I think this also works if you want to type fewer chars 😄  
```suggestion
  optionComparisonKey: 'value' as const,
```